src/ws/test-server.c

```diff
@@ -774,8 +776,19 @@ main (int argc,
   /* avoid gvfs (http://bugzilla.gnome.org/show_bug.cgi?id=526454) */
   g_setenv ("GIO_USE_VFS", "local", TRUE);
 
+  /* playground config and data directories */
+  config_dir = g_dir_make_tmp ("cockpit.config.XXXXXX", NULL);
+  g_assert (config_dir);
+  machines_dir = g_build_filename (config_dir, "cockpit", "machines.d", NULL);
+  g_assert (g_mkdir_with_parents (machines_dir, 0755) == 0);
+  g_free (machines_dir);
+  data_dir = g_dir_make_tmp ("cockpit.data.XXXXXX", NULL);
+  g_assert (data_dir);
+
   g_setenv ("XDG_DATA_HOME", SRCDIR "/src/bridge/mock-resource/home", TRUE);
   g_setenv ("XDG_DATA_DIRS", SRCDIR "/src/bridge/mock-resource/system", TRUE);
+  g_setenv ("XDG_CONFIG_DIRS", config_dir, TRUE);
+  g_setenv ("COCKPIT_DATA_DIR", data_dir, TRUE);
```

Comment:
Yes, just for verifying that `/var/lib/cockpit/machines.json` still gets read. But I can turn this into an integration test. I'm going to add one for adding the migration from /var to /etc anyway (as soon as this one lands), so after that this wouldn't add much extra overhead.